src/ucp/core/ucp_ep.c

```diff
@@ -1270,9 +1274,13 @@ void ucp_ep_config_init(ucp_worker_h worker, ucp_ep_config_t *config)
                                               config->key.rma_bw_lanes,
                                               UCT_IFACE_FLAG_GET_ZCOPY,
                                               max_rndv_thresh);
-                config->tag.eager           = config->am;
-                config->tag.lane            = lane;
-                config->tag.max_eager_short = config->tag.eager.max_short;
+                config->tag.eager                      = config->am;
+                config->tag.lane                       = lane;
+                config->tag.max_eager_short.memtype_on = config->tag.eager.max_short;
```

Comment:
minor: there is code duplication with line 1224-1228, can be extracted to a helper function